include/openPMD/RecordComponent.hpp

```diff
@@ -114,6 +116,14 @@ class RecordComponent : public BaseRecordComponent
      *
      * If offset is non-zero and extent is {-1u} the leftover extent in the
      * record component will be selected.
+     *
+     * In FlushType::DEFER, the data managed by the returned
+     * std::shared_ptr< T > MUST NOT be read, modified or deleted until
+     * Series::flush is called.
+     *
+     * @todo implement: if the shared pointers' `use_count` drops to one
+     *       before Series::flush is called, the load will be skipped during
+     *       Series::flush.
```

Comment:
Whoop! :raised_hands:  This is a great optimization.

Reply:
Yes, but really needs #470 to work :)